locked-asset/simple-lock-energy/src/lock_options.rs

```diff
@@ -70,9 +70,9 @@ pub trait LockOptionsModule {
         );
     }
 
-    fn lock_epoch_to_start_of_month(&self, lock_option: Epoch) -> Epoch {
-        let extra_days = lock_option % EPOCHS_PER_MONTH;
-        lock_option - extra_days
+    fn unlock_epoch_to_start_of_month(&self, unlock_epoch: Epoch) -> Epoch {
+        let extra_days = unlock_epoch % EPOCHS_PER_MONTH;
```

Comment:
This is not very accurate (`EPOCHS_PER_MONTH`). 
Why is start of month important here? Do we need precision?

Reply:
This is done so, for example, a token that were to unlock on 5th september, one that would unlock on 10th, and one on 12th, will all unlock on 1st, hence have the same `token_nonce` for the locked token (since they'd have the same attributes).

This approximation is done so users don't fill their wallets with tokens.